src/coreclr/jit/lsraarm64.cpp

```diff
@@ -1634,15 +1635,32 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree, int* pDstCou
                         {
                             predMask = RBM_LOWMASK.GetPredicateRegSet();
                         }
+
+                        // Special-case, CreateBreakPropagateMask's op2 is the RMW node.
+                        if (intrinEmb.id == NI_Sve_CreateBreakPropagateMask)
+                        {
+                            assert(embOp2Node->isRMWHWIntrinsic(compiler));
+                            assert(!tgtPrefOp1);
+                            assert(!tgtPrefOp2);
+                            assert(!tgtPrefEmbOp2OfOp2);
+                            tgtPrefEmbOp2OfOp2 = true;
+                        }
                     }
                 }
                 else if (HWIntrinsicInfo::IsLowMaskedOperation(intrin.id))
                 {
                     predMask = RBM_LOWMASK.GetPredicateRegSet();
                 }
 
-                if (tgtPrefOp2)
+                if (tgtPrefEmbOp2OfOp2)
+                {
+                    assert(!tgtPrefOp1);
+                    assert(!tgtPrefOp2);
+                    srcCount += BuildDelayFreeUses(intrin.op1, intrin.op2->AsHWIntrinsic()->Op(2));
+                }
+                else if (tgtPrefOp2)
                 {
+                    assert(!tgtPrefOp1);
                     srcCount += BuildDelayFreeUses(intrin.op1, intrin.op2, predMask);
```

Comment:
btw, I just realized that in #104002 this should have been:

```suggestion
                    srcCount += BuildDelayFreeUses(intrin.op2, intrin.op1, predMask);
```

Reply:
It should actually be fine? `intrin.op2` is the RMW node, and BuildDelayFreeUses' second parameter is for the RMW node.

Reply:
No, if you see the summary docs of `BuildDelayFreeUses()`, `node` is the node for which we want to create the `RefPosition` for, in this case `intrin.op2` and `rmwNode` is the node for which we want to make sure that the register doesn't collide with, in this case, it is `intrin.op1`. See some examples of places where we call `BuildDelayFreeUses()`.

```
//------------------------------------------------------------------------
// BuildDelayFreeUses: Build Use RefPositions for an operand that might be contained,
//                     and which may need to be marked delayRegFree
//
// Arguments:
//    node              - The node of interest
//    rmwNode           - The node that has RMW semantics (if applicable)
//    candidates        - The set of candidates for the uses
//    useRefPositionRef - If a use RefPosition is created, returns it. If none created, sets it to nullptr.
//
// REVIEW: useRefPositionRef is not consistently set. Also, sometimes this function creates multiple RefPositions
// but can only return one. Does it matter which one gets returned?
//
// Return Value:
//    The number of source registers used by the *parent* of this node.
//
int LinearScan::BuildDelayFreeUses(GenTree*         node,
                                   GenTree*         rmwNode,
                                   SingleTypeRegSet candidates,
                                   RefPosition**    useRefPositionRef)
```

Reply:
If I make that change, I get the assertion:
> Errors:

Assert failure(PID 15364 [0x00003c04], Thread: 6432 [0x1920]): Assertion failed '!"removeListNode didn't find the node"' in 'JIT.HardwareIntrinsics.Arm._Sve.SimpleBinaryOpTest__Sve_CreateMaskForFirstActiveElement_sbyte:RunBasicScenario_UnsafeRead():this' during 'Linear scan register alloc' (IL size 110; hash 0xd48bee8e; FullOpts)

    File: C:\work\runtime\src\coreclr\jit\lsrabuild.cpp:69
    Image: C:\Users\dotnetperfuser\Desktop\work\windows.arm64.Checked\Tests\Core_Root\corerun.exe

Reply:
fixed it properly.